Lib/test/libregrtest/single.py

```diff
@@ -274,6 +275,9 @@ def _runtest(result: TestResult, runtests: RunTests) -> None:
     output_on_failure = runtests.output_on_failure
     timeout = runtests.timeout
 
+    if can_colorize(file=sys.stdout):
+        os.environ['FORCE_COLOR'] = "1"
```

Comment:
Is this change also needed?

Reply:
Yes for single-process mode coloring of `unittest` output because `unittest` gets given a `io.StringIO` as its `stdout` (on which `isatty` returns False). `./python -m test -W --single-process` with a failing test output is uncolored without this, colored with it.

Reply:
I've been pondering if this would be better done by setting `os.environ['FORCE_COLOR']` inside `libregrtest` main (https://github.com/python/cpython/blob/1e1f43519605b7c54a96aa44d0feed09d2bb1a67/Lib/test/libregrtest/main.py#L755-L779) rather than inside the the single and parallel code paths separately; that seems like it's covering too much code though